### ISSUE TITLE
identify if a installation is running in Kubernetes

### ISSUE THREAD

**Issue by 111andre111** [elastic/support-diagnostics#384]
This seems to be related to issue https://github.com/elastic/support-diagnostics/issues/334

At the moment at the moment, there is no identification in the support diagnostics if a container is running in docker or kubernetes.
It would be extremely helpful anyways to identify if a node is running inside a kubernetes pod.

In order to see if there is a cluster executed in a kubernetes cluster
These variables will be exposed to every pod.
https://kubernetes.io/docs/concepts/services-networking/connect-applications-service/#environment-variables
```
KUBERNETES_SERVICE_HOST
KUBERNETES_SERVICE_PORT
```
We have already docker enablement in place.
So we just can do a 
`docker exec -it <containerid> printenv | grep KUBERNETES_SERVICE`
So if it returns something like that:
```
KUBERNETES_SERVICE_HOST=10.96.0.1
KUBERNETES_SERVICE_PORT=443
KUBERNETES_SERVICE_PORT_HTTPS=443
```
we know that this container is running under Kubernetes.

**Comment by nemonster** [comment]
> At the moment at the moment, there is no identification in the support diagnostics if a container is running in docker or kubernetes.

That's actually only half right. It does check for whether a node is in a Docker container via the process id. If you check the current stats tool there's a column for this. Any calls you make to a container are only going to apply if you are running on a host with Docker containers running. And if you're in Kubernates it's not even necessarily the same host so there's a limit to the usefulness of attempting to go too deep.

I would expect the ECK team to be providing tools to capture more of this information - I don't necessarily think it falls within scope to attempt something this broad for vanilla setups outside of that. Unless they want to do it of course.

I will add a call to the set of Docker calls I already run to see if Kuberates is present. But that's a single, one line works or it doesn't execution. The above is way more work than needs to be done to simply ascertain if Kubernates is loaded on the host being queried.

**Comment by 111andre111** [comment]
Yes, that's right about the docker topic. But not every kubernetes environment will be a vanilla ECK installation out there. So having these environment variables would be extremely helpful already to get a direction idea about what steps need to be done next, when analyzing a topic.
